hibernate-core/src/main/java/org/hibernate/hql/internal/ast/QueryTranslatorImpl.java

```diff
@@ -377,7 +379,16 @@ public List list(SharedSessionContractImplementor session, QueryParameters query
 
 		QueryParameters queryParametersToUse;
 		if ( hasLimit && containsCollectionFetches() ) {
-			LOG.firstOrMaxResultsSpecifiedWithCollectionFetch();
+			boolean fail = ConfigurationHelper.getBoolean(Environment.FAIL_ON_INTENSIVE_IN_MEMORY_OPERATION,
```

Comment:
You can't do this though.  This ends up being a performance problem.  You end up doing a Map lookup every time a query is translated which the JBoss performance team has consistently identified as a problem - and its unnecessary. 

This really should be integrated with SessionFactoryOptions.  I apologize that doing that is currently pretty painful

Reply:
Done !